iOS document (#76)
mirrorworld-universe/docs.mirrorworld.fun: data/ios/ios-api.mdx

# iOS API Reference

## Authentication Methods

`public let MWSDK = MirrorWorldSDK.share`

## StartLogin

> Calling this api would popup a dialog, user can finish login flow on it. In which dialog, user can login with third method like google, twitter. Or he can login with his email which registered on our website.

Reviewer: ```suggestion
Calling this API would popup a dialog, user can finish login flow on it. In which the user can login with another method like Google or Twitter. Or they can login with their registered email.
```

Author: Okay, I'll fix it today.

```Swift
MirrorWorldSDK.share.StartLogin { userInfo in
    print("login success :\(userInfo?.toString() ?? "")")
} onFail: {
    print("login failed !")
}
```

## CheckAuthenticated

> Checks whether the current user is logged in. You can use this function to judge whether a user needs to start login flow.

Reviewer: ```suggestion
Checks whether the current user is logged in. You can use this function to check if a user needs to start the login flow.
```

```Swift
MWSDK.CheckAuthenticated { onBool in
    print("This device's login state is:\(onBool)")
}
```

### Logs out a user

```Swift
MWSDK.loginOut {
    print("Logs out a user : success")
} onFail: {
    print("Logs out a user : failed !")
}
```

## Wallet Methods

### OpenWallet

> Open a webview which would show the wallet page.

Reviewer: ```suggestion
Opens a web view which shows the wallet page.
```

```Swift
MWSDK.OpenWallet()
```

### GetAccessToken

> Get access token so that users can visit APIs.

Reviewer: ```suggestion
Gets access token for users to use the API.
```
